document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -132,6 +134,44 @@ public boolean upsert(Key key, Document document) throws IOException {
     }
   }
 
+  /**
+   * Bulk updates existing documents if condition for the corresponding document evaluates to true.
+   */
+  @Override
+  public boolean bulkUpdate(List<Triple<Key, Document, Filter>> documents) {
+    try {
+      BulkWriteResult result = bulkUpdateImpl(documents);
+      LOGGER.debug(result.toString());
+      return true;
```

Comment:
I think for now, we should return `result.getModifiedCount()`. Later, we may want to return the list of IDs that got updated (or not updated).

Reply:
done